Modules/_datetimemodule.c

```diff
@@ -6782,17 +6800,18 @@ _datetime_exec(PyObject *module)
         return -1;
     }
 
-    PyObject *x = create_timezone(delta, NULL);
+    PyObject *utc = create_timezone(delta, NULL);
     Py_DECREF(delta);
-    if (x == NULL) {
+    if (utc == NULL) {
         return -1;
     }
-    if (PyDict_SetItemString(d, "utc", x) < 0) {
-        Py_DECREF(x);
+    if (PyDict_SetItemString(d, "utc", utc) < 0) {
+        Py_DECREF(utc);
         return -1;
     }
 
-    PyDateTime_TimeZone_UTC = x;
+    datetime_state *st = GLOBAL_STATE();
+    st->utc = utc;
```

Comment:
Please add `utc = NULL` to make it more explicit that st->utc becomes the new owner the reference.

Reply:
Better to use `Py_SETREF`.

```suggestion
    Py_SETREF(st->utc, utc);
```

Reply:
Oh, of course; that won't work 😆

Reply:
Fixed again.